sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClientOptions.cs

```diff
@@ -50,5 +51,19 @@ internal static string GetVersionString(ServiceVersion version)
                 _ => throw new NotSupportedException($"The service version {version} is not supported."),
             };
         }
+
+        /// <summary>
+        /// Add headers that are considered safe for logging or including in error messages
+        /// by default.
+        /// </summary>
+        private void AddLoggedHeaders()
```

Comment:
We tend to do this in a separate class `ClientOptionsExtensions`
See:
- [TA](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/ClientOptionsExtensions.cs)
- [Appconfig](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ClientOptionsExtensions.cs)

Reply:
I don't think there's really a pattern here. [Storage Blobs](https://github.com/Azure/azure-sdk-for-net/blob/2f5c475ab87b045ca56e1352272f046ef8701b02/sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs#L99) and [Search](https://github.com/Azure/azure-sdk-for-net/blob/2f5c475ab87b045ca56e1352272f046ef8701b02/sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs#L101) put them in the `Options`, like I did. Personally I prefer keeping it `private` to creating a new extension class, unless we intend to use it in other places.

Reply:
o.O interesting. ok